### ISSUE TITLE
Improved Fill Types

### ISSUE THREAD

**Issue by ranjian0** [ranjian0/building_tool#39]
This issue is the base for work to begin on improving fill types. We would like to add a couple more fill types and also provide a set of quick presets for common fill types and/or combinations of fill types. The image references below should act as a guide.

## Window
![Imgur](https://i.imgur.com/hweB1Z1.jpg)
![Imgur](https://i.imgur.com/1e8f5wd.jpg)

## Door
![Imgur](https://i.imgur.com/UeIcaQT.gif)

While there is no clear cut plan, the above images give a good indication of the direction we should be heading. New issues concerning this should refer here.

A complete task list may be added later.

**Comment by MCrafterzz** [comment]
Window boards would also be cool.

**Comment by ranjian0** [comment]
> Window boards would also be cool.

Illustrations for this would be really handy. Just drop them here!

**Comment by ranjian0** [comment]
In progress Fill Types:
- [ ] Diamond
- [ ] Cross Panel
- [ ] Full Louver
- [ ] Upper Louver
- [ ] Lower Louver
- [ ] Split Louver
- [ ] Half Glass
- [ ] Half Glass 1 panel
- [ ] Half Glass 2 panel
- [ ] Half Glass 3 panel
- [ ] Half Glass 4 panel
- [ ] Half Glass 2 panel horizontal
- [ ] Half Glass 3 panel horizontal

**Comment by luckykadam** [comment]
It would be useful to have an option for `bottom_panel` in doors.

Behaviour:

1. Enabling bottom panel will divide the door in two panels (height of bottom panel is configurable)
2. Bottom panel can have its own fill type
3. Use cases
    Top panel - glass panes, Bottom panel - panels
    Top panel - vertical panels, Bottom panel - horizontal panels

@ranjian0 What do you think ?

**Comment by ranjian0** [comment]
> It would be useful to have an option for `bottom_panel` in doors.
> 
> Behaviour:
> 
>     1. Enabling bottom panel will divide the door in two panels (height of bottom panel is configurable)
> 
>     2. Bottom panel can have its own fill type
> 
>     3. Use cases
>        Top panel - glass panes, Bottom panel - panels
>        Top panel - vertical panels, Bottom panel - horizontal panels
> 
> 
> @ranjian0 What do you think ?

Yep. My thoughts exactly. Was just thinking of half fill feature similar to your idea. Configurable height is a really good option too.

**Comment by MCrafterzz** [comment]
> > It would be useful to have an option for `bottom_panel` in doors.
> > Behaviour:
> > ```
> > 1. Enabling bottom panel will divide the door in two panels (height of bottom panel is configurable)
> > 
> > 2. Bottom panel can have its own fill type
> > 
> > 3. Use cases
> >    Top panel - glass panes, Bottom panel - panels
> >    Top panel - vertical panels, Bottom panel - horizontal panels
> > ```
> > 
> > 
> > @ranjian0 What do you think ?
> 
> Yep. My thoughts exactly. Was just thinking of half fill feature similar to your idea. Configurable height is a really good option too.

I agree, the question is how flexible the system should be, either you just allow having two sections or you allow it to be any numbers. If you had two panels wide and 4 panels high then you could create the 8 panel door from the picture, just something to think about

**Comment by ranjian0** [comment]
I think two should be enough. If more is needed, we can implement it in such a way that fills can be added recursively. e.g
You make an initial fill with two sections but leave one of them empty and then use the fill tool again on the empty section.

**Comment by MCrafterzz** [comment]
> I think two should be enough. If more is needed, we can implement it in such a way that fills can be added recursively. e.g
> You make an initial fill with two sections but leave one of them empty and then use the fill tool again on the empty section.
> 
> Possibilities are endless.

Thats really smart, typical situation of when you can simply things as its in blender :P As long as you can change where the door/window divides and if it should be done vertically or horizontally then you could do anything
